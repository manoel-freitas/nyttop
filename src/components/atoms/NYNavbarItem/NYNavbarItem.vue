<template>
  <router-link tag="li" :class="itemClass" :to="to">
    <button class="button is-white">{{ label }}</button>
  </router-link>
</template>

<script>
export default {
  props: {
    to: {
      type: Object,
      required: true
    },
    label: {
      type: String,
      required: true
    },
    itemClass: {
      type: String,
      required: true
    }
  }
}
</script>
