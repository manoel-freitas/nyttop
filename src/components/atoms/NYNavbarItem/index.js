import NYNavbarItem from './NYNavbarItem'

export default NYNavbarItem
