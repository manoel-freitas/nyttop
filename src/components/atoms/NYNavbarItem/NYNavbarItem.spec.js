/// <reference types="jest" />

import { mount } from '@vue/test-utils'
import NYNavbarItem from './NYNavbarItem'

const nyNavbarItem = mount(NYNavbarItem, {
  propsData: {
    to: {
      name: 'sci-tec'
    },
    label: 'Ciencias e tecnologia',
    itemClass: 'levelbar-item'
  },
  stubs: ['router-link']
})
test('NYNavbarItem has to be mounted', () => {
  expect(nyNavbarItem.isVueInstance()).toBeTruthy()
})

test('snapshot test', () => {
  expect(nyNavbarItem.element).toMatchSnapshot()
})
