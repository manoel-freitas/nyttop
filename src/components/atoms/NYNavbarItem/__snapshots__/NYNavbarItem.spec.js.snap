// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<router-link-stub
  class="levelbar-item"
  tag="li"
  to="[object Object]"
>
  <button
    class="button is-white"
  >
    Ciencias e tecnologia
  </button>
</router-link-stub>
`;
